web-console/src/lib/components/streaming/management/PipelineTable.tsx

```diff
@@ -601,7 +581,9 @@ const PipelineActions = (params: { row: Pipeline }) => {
   const deletePipelineClick = useDeletePipeline()
 
   const pipelineStatus = usePipelineStateStore(state => state.clientStatus)
-  const curProgramQuery = useQuery<ProgramDescr>(['programCode', { program_id: pipeline.program_id }], {
+  invariant(pipeline.program_id, 'PipelineActions: missing program_id!')
```

Comment:
I think this invariant doesn't hold. program_id can be null if you just make a pipeline and don't set a program yet, see also the enabled condition ` enabled: pipeline.program_id != null` on row 587

Reply:
Absolutely